src/components/articles/UpdateArticle.jsx

```diff
@@ -0,0 +1,151 @@
+import { clearError } from "../../redux/actions/common";
+
+class UpdateArticle extends React.Component {
+  state = {
+    title: "",
+    description: "",
+    body: "",
+    loaded: false
+  };
+
+  componentDidMount() {
+    const { match, dispatch } = this.props;
+    dispatch(viewSingleArticle(match.params.slug));
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { loaded } = this.state;
+    if (!loaded) {
+      this.setState(state => ({
+        ...state,
+        loaded: true,
+        title: nextProps.article.Article.title,
+        description: nextProps.article.Article.description,
+        body: nextProps.article.Article.body
+      }));
+    }
+  }
+
+  handleChange = e => {
+    const { value, name } = e.target;
+    const { dispatch } = this.props;
+
+    this.setState(state => ({ ...state, [name]: value }));
+    dispatch(clearError());
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    const { title, body, description } = this.state;
+    const { dispatch, match } = this.props;
+    dispatch(
+      updateArticle(match.params.slug, {
+        title,
+        body,
+        description
+      })
+    );
+  };
+
+  render() {
+    const { title, body, description } = this.state;
+    const { updating, error } = this.props;
+    return (
+      <div className="container">
+        <div className="d-flex justify-content-end" />
+        <div className="row ah-editor-container">
+          <div className="ml-auto mr-auto col-sm-8 p-4">
+            <div>
+              <form onSubmit={this.handleSubmit} className="mb-4">
+                <TextInput
+                  type="text"
+                  placeholder="Title"
+                  name="title"
+                  value={title}
+                  onChange={this.handleChange}
+                  className={
+                    error.title
+                      ? "form-control error ah-editor-field ah-article-title"
+                      : "form-control ah-editor-field ah-article-title"
+                  }
+                  error={error.title}
+                  required
+                />
+                <TextInput
+                  type="text"
+                  placeholder="A cool subtitle"
+                  name="description"
+                  value={description}
+                  onChange={this.handleChange}
+                  className={
+                    error.description
+                      ? "error form-control ah-editor-field ah-article-description"
+                      : "form-control ah-editor-field ah-article-description"
+                  }
+                  error={error.description}
+                  required
+                />
+                <TextArea
+                  placeholder="Write your story"
+                  name="body"
+                  value={body}
+                  onChange={this.handleChange}
+                  className={
+                    error.body
```

Comment:
Nice job on the arrangement. Improvement.

Reply:
Thank you!